torchinductor/lowering.py

```diff
@@ -96,6 +114,20 @@ def decode_device(device):
     return device
 
 
+def get_promoted_dtype(*args):
+    # TODO: fix other dtype places in this file
+    def construct_input(inp):
+        if isinstance(inp, Number):
+            return inp
+        else:
+            assert hasattr(inp, "get_dtype")
+            dim = len(inp.get_size())
+            # construct a tmp tensor to feed into torch.result_type
+            return torch.zeros([1] * dim, dtype=inp.get_dtype())
+
+    return functools.reduce(torch.result_type, [construct_input(arg) for arg in args])
```

Comment:
Unfortunately, this won't be correct for more than 2 args (because type promotion is not associative), but we expect to call it for 2 args only in lowerings, so that should be ok.